juicy-highlight.html

```diff
@@ -1,294 +1,291 @@
                 }
 
-                if (el.style.height != h) {
-                    el.style.height = h;
+                if (hidden || maskElement.currentRectSvg != rectSvg) {
+                    rect.setAttribute("d", rectSvg);
+                    maskElement.currentRectSvg = rectSvg;
                 }
-            }
-        }
 
-        JuicyHighlightPrototype.show = function (elements, getContent) {
-            if (Array.isArray(elements)) {
-                this.focused = elements;
-            }
-            else if (elements instanceof NodeList) {
-                this.focused = Array.prototype.slice.call(elements); //convert NodeList to Array
-            }
-            else if (elements) {
-                this.focused = [elements];
-            }
-            else {
-                this.focused = [];
-            }
+                const w = availableWidth + "px";
+                const h = availableHeight + "px";
+                const els = [svg, focusRect];
 
-            this.generateContent(getContent);
+                for (let i = 0; i < els.length; i++) {
+                    const el = els[i];
 
-            if (!this.overlayRect) {
-                this.overlayRect = createMask("#555555", 0.75);
-            }
-            if (!this.focusRect) {
-                this.focusRect = createFocusRect(this.strokeWidth, this.strokeColor, this.fill);
-            }
-            if (this.overlay && !this.overlayRect.parentNode) {
-                document.body.appendChild(this.overlayRect);
+                    if (el.style.width != w) {
+                        el.style.width = w;
+                    }
+
+                    if (el.style.height != h) {
+                        el.style.height = h;
+                    }
+                }
             }
-            if (this.focusRect && !this.focusRect.parentNode) {
-                document.body.appendChild(this.focusRect);
+            
+        const attributes = ["overlay", "strokewidth", "strokeoffset", "strokecolor", "fill"];
+        
+        class JuicyHighlightElement extends HTMLElement {            
+            constructor(self) {
+                self = super(self);
+                self.overlay = false;
+                self.fill = false;
+                self.strokeWidth = 1;
+                self.strokeOffset = 1;
+                self.strokeColor = "#000000";
+                self.focused = null;
+                self.contents = [];
+                self.currentState = null;
+                self.currentMaskSvg = null;
+                self.currentRectSvg = null;
+
+                for (let i = 0; i < attributes.length; i++) {
+                    if (self.hasAttribute(attributes[i])) {
+                        self.attributeChangedCallback(attributes[i], null, self.attributes[attributes[i]].value);
+                    }
+                }
+            };
+            
+            static get observedAttributes() {
+                return attributes; 
```

Comment:
That's leaky, you are exposing the private attributes variable: http://jsbin.com/debotey/1/edit?js,console